hist/hist/inc/TNDArray.h

```diff
@@ -128,31 +121,25 @@ class TNDArrayT: public TNDArray {
    fNumData(), fData() {
       fNumData = fSizes[0];
    }
-   ~TNDArrayT() {
-      delete[] fData;
-   }
+   ~TNDArrayT() {}
 
    void Init(Int_t ndim, const Int_t* nbins, bool addOverflow = false) {
-      delete[] fData;
-      fData = 0;
+      fData.clear();
       TNDArray::Init(ndim, nbins, addOverflow);
       fNumData = fSizes[0];
    }
 
    void Reset(Option_t* /*option*/ = "") {
       // Reset the content
-
-      // Use placement-new with value initialization:
-      if (fData) {
-         new (fData) T[fNumData]();
-      }
+      fData.clear();
+      fData.resize(fNumData, T());
```

Comment:
```suggestion
      fData.assign(fNumData, T());
```
(all occurrences, please)